Model/User.hs

```diff
@@ -45,11 +53,32 @@ upsertUser user = entityKey <$> upsert user
     , UserEmail =. userEmail user
     ]
 
-credsToUser :: Maybe PlanId -> Creds m -> Maybe User
+credsToUser :: Maybe PlanId -> Creds m -> Either String User
 credsToUser mplanId Creds{..} = User
-    <$> lookup "name" credsExtra
-    <*> lookup "email" credsExtra
+    <$> (profileName <$> eprofile)
+    <*> (profileEmail <$> eprofile)
     <*> pure credsPlugin
     <*> pure credsIdent
-    <*> mplanId
+    <*> eplanId
     <*> pure Nothing
+
+  where
+    eplanId = maybe (Left "Free plan not found") Right mplanId
```

Comment:
I wonder if it would be better to do the `Maybe` to `Either` conversion in `authenticateUser`, since this function doesn't actually look for the free plan.